embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/Device.kt

```diff
@@ -0,0 +1,286 @@
+package io.embrace.android.embracesdk.capture.envelope.resource
+
+import android.os.Build
+import android.os.Environment
+import android.os.StatFs
+import android.util.DisplayMetrics
+import android.view.WindowManager
+import androidx.annotation.ChecksSdkIntAtLeast
+import io.embrace.android.embracesdk.capture.cpu.CpuInfoDelegate
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logDebug
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logDeveloper
+import io.embrace.android.embracesdk.prefs.PreferencesService
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import java.io.File
+import java.util.Locale
+
+public interface Device {
+
+    /**
+     * Tries to determine whether the device is jailbroken by looking for specific directories which
+     * exist on jailbroken devices. Emulators are excluded and will always return false.
+     *
+     * @return true if the device is jailbroken and not an emulator, false otherwise
+     */
+    public var isJailbroken: Boolean?
+
+    /**
+     * Gets the device's screen resolution.
+     *
+     * @param windowManager the {@link WindowManager} from the {@link Context}
+     * @return the device's screen resolution
+     */
+    public var screenResolution: String
+
+    /**
+     * Gets the name of the manufacturer of the device.
+     *
+     * @return the name of the device manufacturer
+     */
+    public val manufacturer: String?
+
+    /**
+     * Gets the name of the model of the device.
+     *
+     * @return the name of the model of the device
+     */
+    public val model: String?
+
+    /**
+     * Gets the operating system of the device. This is hard-coded to Android OS.
+     *
+     * @return the device's operating system
+     */
+    public val operatingSystemType: String
+
+    /**
+     * Gets the version of the installed operating system on the device.
+     *
+     * @return the version of the operating system
+     */
+    public val operatingSystemVersion: String?
+
+    /**
+     * Gets the version code of the running Android SDK.
+     *
+     * @return the running Android SDK version code
+     */
+    public val operatingSystemVersionCode: Int
+
+    /**
+     * Get the number of available cores for device info
+     *
+     * @return Number of cores in long
+     */
+    public val numberOfCores: Int
+
+    /**
+     * Gets the free capacity of the internal storage of the device.
+     *
+     * @param statFs the {@link StatFs} service for the device
+     * @return the total free capacity of the internal storage of the device in bytes
+     */
+    public val internalStorageTotalCapacity: Lazy<Long>
```

Comment:
nit: this should probably be type `Long` in the interface, and the implementer can choose to implement it as `Lazy`. Not a big deal but it'll make fakes marginally easier to write